components/google_drive/actions/add-file-sharing-preference/add-file-sharing-preference.js

```diff
@@ -0,0 +1,96 @@
+const googleDrive = require("../../google_drive.app");
+
+module.exports = {
+  key: "google_drive-add-file-sharing-preference",
+  name: "Add File Sharing Preference",
+  description: "Add File Sharing Preference",
+  version: "0.0.12",
+  type: "action",
+  props: {
+    googleDrive,
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description: "The drive you want to find a file in",
+      optional: true,
```

Comment:
What happens if the user doesn't provide a drive? Shouldn't this be required, since the `fileId` depends on this prop being set?

Reply:
If the user doesn't provide a drive, `files.list` searches files in the `user` [corpus](https://developers.google.com/drive/api/v3/search-files#search_the_corpora) by default, which [includes](https://developers.google.com/drive/api/v3/about-files#org) all files in "My Drive" and "Shared with me."